### PR TITLE
Add gid to worker, make uid/gid int

### PR DESCRIPTION
**Pull request by athornton**
If the auth system doesn't provide uid/gid for users (roughly speaking, if the auth system isn't LDAP-backed) the noteburst system needs to provide both: the Nublado controller needs both a uid and a gid in order to spawn a user.

At some time, uid became a string; according to the (unused, presumably) values-minikube.yaml for Noteburst it used to be an int.  Since uid and gid are Unix account concepts, they're both properly ints rather than strings.

This PR implements that change in the code and test harness.

A different approach would be to drop those fields entirely, and just insist that the auth system do the work for us.  This will work at IDF and USDF, anyway, which is probably all we really care about.

### CODE REVIEW

**Review by jonathansick — APPROVED** [review]
The changes look good, thanks. I'm curious to know from @rra what might have changed in the system so that both uid and gid are necessary in non-LDAP systems when it wasn't before when we used GitHub auth?

**Comment by jonathansick** [comment]
Please add a change log entry with scriv before merging.

**Comment by rra** [comment]
Nublado v2 had a fallback where it blindly assumed the GID was the same as the UID if the Gafaelfawr data didn't contain a GID. When we rewrote that code for Nublado v3, I made the GID required to avoid baking that assumption into a layer of the stack that you wouldn't expect to be allocating GIDs.

I've been thinking about making Gafaelfawr require all users have primary GIDs so that the user creation would cleanly fail in Gafaelfawr, but have been holding off just because Nublado is the only service that cares and if people deploy Phalanx somewhere without a POSIX file system, they may not want to go to the work of creating a GID. Probably the right answer is to add a configuration parameter to Gafaelfawr to say whether a GID is required for a given environment.

**Pull request merged**
